UI/ViewModels/MockEditViewModel.cs

```diff
@@ -0,0 +1,244 @@
+            BroadcastMockTemplateUpdated();
+        }
+    }
+
+    /// <summary>
+    /// Topic to send Mock
+    /// </summary>
+    public string Topic
+    {
+        get => _topic;
+        set
+        {
+            this.RaiseAndSetIfChanged(ref _topic, value);
+            BroadcastMockTemplateUpdated();
+        }
+    }
+
+    public string? ValidationError
+    {
+        get => _validationError;
+        set => this.RaiseAndSetIfChanged(ref _validationError, value);
+    }
+
+    /// <summary>
+    /// Mock body
+    /// </summary>
+    public string AnswerTemplate
+    {
+        get => _answerTemplate;
+        set
+        {
+            this.RaiseAndSetIfChanged(ref _answerTemplate, value);
+            BroadcastMockTemplateUpdated();
+        }
+    }
+
+    public MockTypes MockType
+    {
+        get => _MockType;
+        set
+        {
+            this.RaiseAndSetIfChanged(ref _MockType, value);
+            BroadcastMockTemplateUpdated();
+        }
+    }
+
+    public bool ShowReplySection
+    {
+        get => _showReplySection;
+        set
+        {
+            this.RaiseAndSetIfChanged(ref _showReplySection, value);
+            MockRowSpan = value
+                ? 1
+                : 2;
+        }
+    }
+
+    public int MockRowSpan
+    {
+        get => _MockRowSpan;
+        set => this.RaiseAndSetIfChanged(ref _MockRowSpan, value);
+    }
+
+    /// <summary>
+    /// Current Mock template data
+    /// </summary>
+    public MockTemplate MockTemplate =>
+        new()
+        {
+            Id = MockId,
+            Name = Name,
+            AnswerTemplate = AnswerTemplate,
+            Topic = Topic,
+            Type = MockType
+        };
+
+    public MockEditViewModel()
+    {
+        MockId = Guid.NewGuid();
+        var scope = Program.Container.BeginLifetimeScope();
+        _mockEngine = scope.Resolve<MockEngine>();
+        _storage = scope.Resolve<IDataStorage>();
+        MockType = MockTypes.SimpleRequestReply;
+
+        Init();
+    }
+
+    public MockEditViewModel(MockTemplate mockTemplate)
+    {
+        Activator = new ViewModelActivator();
+        this.WhenActivated(disposables =>
+        {
+            Disposable
+                .Create(Dispose)
+                .DisposeWith(disposables);
+        });
+
+        var scope = Program.Container.BeginLifetimeScope();
+        _mockEngine = scope.Resolve<MockEngine>();
+        _storage = scope.Resolve<IDataStorage>();
+        MockId = mockTemplate.Id;
+        Name = mockTemplate.Name;
+        Topic = mockTemplate.Topic;
+        AnswerTemplate = mockTemplate.AnswerTemplate;
+        MockType = mockTemplate.Type;
+
+        Init();
+    }
+
+    private void Init()
+    {
+        EnableMock = ReactiveCommand.CreateFromTask(async _ =>
+        {
+            ValidationError = ValidateForm();
+            if (ValidationError != null)
+                return;
+            try
+            {
+                switch (MockType)
```

Comment:
No need for `switch` here. MockEngine already doing all this work. Just pass Template